litmus-portal/authentication/Dockerfile

```diff
@@ -1,10 +1,16 @@
 # BUILD STAGE
 FROM golang:1.14 AS builder
 
+ARG TARGETOS=linux
```

Comment:
Instead of hardcoding TARGETOS=linux. Can we do this? Let the user set the environment from the GitHub workflow

```
ARG TARGETPLATFORM

RUN export GOOS=$(echo ${TARGETPLATFORM} | cut -d / -f1) && \ 
    export GOARCH=$(echo ${TARGETPLATFORM} | cut -d / -f2)
```

Reply:
We have assigned a default value to `TARGETOS` ENV if nothing is provided it will take  `TARGETOS` as linux (This is to handle a case when someone is not using buildx and build the image without passing args).

Docker itself set these variables from the `--platform` provided in buildx command or action https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope

Reply:
I was saying TARGETOS is hardcoded, If we can pass it as env. Here is the list of platforms
```
 "darwin/386" "darwin/amd64" "linux/386" "linux/amd64" "linux/arm" "linux/arm64" "windows/386" "windows/amd64" "windows/arm" ..etch
```

Reply:
For now we can pass linux as TARGET_OS

Reply:
Ok, I have now taken the `PLATFORM` as an ENV from GitHub Secret. One can provide all the target platforms as CSV.
![Screenshot from 2021-06-02 21-56-08](https://user-images.githubusercontent.com/35391335/120516790-5d9a7080-c3ed-11eb-8190-d64077e0c20f.png)

Reply:
Don't make it a secret. Put in the environment variable section.

Reply:
TARGETOS is still pointing to linux. You can put these commands

```
ARG TARGETPLATFORM

RUN export GOOS=$(echo ${TARGETPLATFORM} | cut -d / -f1) && \ 
    export GOARCH=$(echo ${TARGETPLATFORM} | cut -d / -f2)
```

Reply:
Have tested with darwin/386 the OS was updating
Logs: https://github.com/uditgaurav/test-litmus/runs/2730208968?check_suite_focus=true#step:6:107